ui/src/sources/containers/SourcePage.js

```diff
@@ -78,57 +84,106 @@ export const SourcePage = React.createClass({
     createSource(newSource)
       .then(({data: sourceFromServer}) => {
         this.props.addSourceAction(sourceFromServer)
-        this.setState({source: sourceFromServer, error: null})
+        this.setState({
+          source: {...DEFAULT_SOURCE, ...sourceFromServer},
+          isCreated: true,
+        })
       })
-      .catch(({data: error}) => {
+      .catch(err => {
         // dont want to flash this until they submit
-        this.setState({error: error.message})
+        const error = this.parseError(err)
+        console.error('Error on source creation: ', error)
       })
-  },
-
-  handleSubmit(newSource) {
-    const {router, params, addFlashMessage} = this.props
-    const {error} = this.state
-
-    if (error) {
-      return addFlashMessage({type: 'error', text: error})
+  }
+
+  handleSubmit = e => {
+    e.preventDefault()
+    const {notify} = this.props
+    const {isCreated, source, editMode} = this.state
+    const isNewSource = !editMode
+
+    if (!isCreated && isNewSource) {
+      return createSource(source)
+        .then(({data: sourceFromServer}) => {
+          this.props.addSourceAction(sourceFromServer)
+          this._redirect(sourceFromServer)
+        })
+        .catch(error => {
+          this.handleError(error)
+        })
     }
 
-    updateSource(newSource)
+    updateSource(source)
       .then(({data: sourceFromServer}) => {
         this.props.updateSourceAction(sourceFromServer)
-        router.push(`/sources/${params.sourceID}/manage-sources`)
-        addFlashMessage({type: 'success', text: 'The source info saved'})
+        this._redirect(sourceFromServer)
+        notify('success', 'The source info saved')
       })
-      .catch(() => {
-        addFlashMessage({
-          type: 'error',
-          text: 'There was a problem updating the source. Check the settings',
-        })
+      .catch(error => {
+        this.handleError(error)
       })
-  },
+  }
+
+  handleError = err => {
+    const {notify} = this.props
+    const error = this.parseError(err)
+    console.error('Error: ', error)
+    notify('error', `There was a problem: ${error}`)
```

Comment:
Is this a user-facing string?  If so, could this give more context? "Unable to create connection to source... or something"